packages/web-runtime/src/components/Topbar/ApplicationsMenu.vue

```diff
@@ -22,48 +22,63 @@
     >
-      <oc-list class="applications-list">
-        <li v-for="(n, nid) in applicationsList" :key="`apps-menu-${nid}`" @click="clickApp(n)">
-          <oc-button
-            :key="n.url ? 'apps-menu-external-link' : 'apps-menu-internal-link'"
-            :type="n.url ? 'a' : 'router-link'"
-            :target="n.target"
-            :href="n.url"
-            :to="n.path"
-            :appearance="n.active ? 'raw-inverse' : 'raw'"
-            :variation="n.active ? 'primary' : 'passive'"
-            :class="{ 'oc-background-primary-gradient router-link-active': n.active }"
-          >
-            <span class="icon-box">
-              <oc-icon :name="n.icon" variation="inherit" />
-            </span>
-            <span v-text="$gettext(n.title)" />
-            <oc-icon v-if="n.active" name="check" class="active-check" variation="inherit" />
-          </oc-button>
-        </li>
-      </oc-list>
+      <div class="oc-display-block oc-position-relative">
+        <oc-list class="applications-list">
+          <li v-for="(n, nid) in applicationsList" :key="`apps-menu-${nid}`">
+            <oc-button
+              :key="n.url ? 'apps-menu-external-link' : 'apps-menu-internal-link'"
+              :type="n.url ? 'a' : 'router-link'"
+              :target="n.target"
+              :href="n.url"
+              :to="n.path"
+              :appearance="n.active ? 'raw-inverse' : 'raw'"
+              :variation="n.active ? 'primary' : 'passive'"
+              :class="{ 'oc-background-primary-gradient router-link-active': n.active }"
+            >
+              <oc-application-icon :key="appIconKey" :icon="n.icon" :color-primary="n.color" />
+              <span v-text="$gettext(n.title)" />
+            </oc-button>
+          </li>
+        </oc-list>
+      </div>
     </oc-drop>
   </nav>
 </template>
 
 <script lang="ts">
-import { defineComponent, PropType, ComponentPublicInstance } from 'vue'
-import { configurationManager } from 'web-pkg/src/configuration'
-import { urlJoin } from 'web-client/src/utils'
+import { defineComponent, PropType, ComponentPublicInstance, ref, computed } from 'vue'
 import { OcDrop } from 'design-system/src/components'
+import OcApplicationIcon from 'design-system/src/components/OcApplicationIcon/OcApplicationIcon.vue'
+import { useGettext } from 'vue3-gettext'
+import * as uuid from 'uuid'
 
 export default defineComponent({
+  components: {
+    OcApplicationIcon
+  },
   props: {
     applicationsList: {
       type: Array as PropType<any[]>,
       required: false,
       default: () => []
     }
   },
-  computed: {
-    applicationSwitcherLabel() {
-      return this.$gettext('Application Switcher')
+  setup() {
+    const { $gettext } = useGettext()
+    const appIconKey = ref('')
+
+    const applicationSwitcherLabel = computed(() => {
+      return $gettext('Application Switcher')
+    })
+    const updateAppIcons = () => {
+      appIconKey.value = uuid.v4().replaceAll('-', '')
```

Comment:
It's not quite what I meant, now the key still isn't unique 😅 The idea was to give `oc-application-icon` a `key` like `${nid}-${actionKeySuffix}`. The `actionKeySuffix` will then be initialized with an empty ref and re-generated in this method.

Reply:
aah yes right